monkey/infection_monkey/network/relay/tcp_connection_handler.py

```diff
@@ -0,0 +1,41 @@
+import socket
+from threading import Event, Thread
+from typing import Callable, List
+
+PROXY_TIMEOUT = 2.5
+
+
+class TCPConnectionHandler(Thread):
+    """Accepts connections on a TCP socket."""
+
+    def __init__(
+        self,
+        local_port: int,
+        local_host: str = "",
+        client_connected: List[Callable[[socket.socket], None]] = [],
+    ):
+        self.local_port = local_port
+        self.local_host = local_host
+        self._client_connected = client_connected
+        super().__init__(name="TCPConnectionHandler", daemon=True)
+        self._stopped = Event()
```

Comment:
We use this pattern (`self._stopped`)in more than one class. I think `self._interrupt` is more consistent with language in other parts of the agent. Additionally, can we create an `InterruptibleThreadMixin` in `infection_monkey.utils.threading` and reduce duplication?